### DISCUSSION TITLE
how to customize rich.progress.track progress indicators?

### DISCUSSION THREAD

**Discussion by sandrotosi** [Textualize/rich#2693] [General] [answered]
Hello,
i'm using rich.progress.track to track the progress of some long-running tasks, and i'd like to be able to customize the information displayed:

* instead of a percentage, i would like a `nn/total` counter (which also implicitly tells me how many items i have to process)
* at the end of the execution, the time counter shows `0:00:00` while i would like to know how long it took to complete, so `0:12:34` if it took 12 minutes and 34 secs to run

i looked at the doc and the code, and it seems this is not supported directly. Is this something you'd consider adding to rich? is there a way to obtain this already with what rich provides?

thanks for writing this great library! :)

**Comment by willmcgugan** [comment, marked as answer]
Have a look at the docs on [progress columns](https://rich.readthedocs.io/en/latest/progress.html#columns)

There is an `MofNCompleteColumn` which I think does your first point. And the `TimeElapsedColumn` has a `elapsed_when_finished` boolean which does your second point.

**Comment by sandrotosi** [comment]
fantastic, that's what i needed!

small correction: it's `TimeRemainingColumn` that has the `elapsed_when_finished` argument :)
